src/Repository/StatsContentRepository.php

```diff
@@ -177,106 +121,72 @@ public function getStats(
             default:
                 throw new \LogicException('Invalid interval provided');
         }
+        if (null !== $start && null === $end) {
+            $end = $start->modify('-1 '.$interval);
+        } elseif (null === $start && null !== $end) {
+            $start = $end->modify('+1 '.$interval);
+        }
 
-        $this->start = $start ?? new \DateTime('-1 '.$interval);
-
-        $toReturn = [
-            'entry' => $this->aggregateStats('entry', $magazine, $interval, $end),
-            'entry_comment' => $this->aggregateStats('entry_comment', $magazine, $interval, $end),
-            'post' => $this->aggregateStats('post', $magazine, $interval, $end),
-            'post_comment' => $this->aggregateStats('post_comment', $magazine, $interval, $end),
+        return [
+            'entry' => $this->aggregateStats('entry', $start, $end, true !== $onlyLocal, $magazine),
+            'entry_comment' => $this->aggregateStats('entry_comment', $start, $end, true !== $onlyLocal, $magazine),
+            'post' => $this->aggregateStats('post', $start, $end, true !== $onlyLocal, $magazine),
+            'post_comment' => $this->aggregateStats('post_comment', $start, $end, true !== $onlyLocal, $magazine),
         ];
-
-        return $toReturn;
     }
 
-    private function aggregateStats(string $table, ?Magazine $magazine, string $interval, ?\DateTime $end): array
+    public function aggregateStats(string $tableName, ?\DateTimeImmutable $sinceDate, ?\DateTimeImmutable $tilDate, bool $federated, ?Magazine $magazine): int
     {
-        if (null === $end) {
-            $end = new \DateTime();
+        $tableName = match ($tableName) {
+            'entry' => 'entry',
+            'entry_comment' => 'entry_comment',
+            'post' => 'post',
+            'post_comment' => 'post_comment',
+            default => throw new \InvalidArgumentException("$tableName is not a valid countable")
+        };
+
+        $federatedCond = $federated ? ' AND e.ap_id IS NULL ' : '';
+        $magazineCond = $magazine ? 'AND e.magazine_id = :magId' : '';
+        $SinceDateCond = $sinceDate ? 'AND e.created_at > :date' : '';
```

Comment:
was this meant to be capitalized? it being gold makes me wonder what php thinks of this. I'm used to capital being exported in golang but not familiar with php xD